File: Easy/Q191-NumberOf1Bits.js
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function(n) {
    let count = 0;
    let totalBitsLen = 32;

    while(totalBitsLen--) {
        count += n & 1;
        n = n >> 1;
    }

    return count;
};

File: cpp/Leetcode/BitOperation/Q191-Numberof1Bits.cpp
class Solution {
public:
  int hammingWeight(int n) {
    int count = 0;

    while (n > 0) {
      if (n % 2 == 1)
        ++count;
      n /= 2;
    }

    return count;
  }
};

class SolutionBitOperation {
public:
  int hammingWeight(int n) {
    int count = 0;

    for (int i = 0; i < 32; ++i) {
      if ((n >> i) & 1)
        ++count;
    }

    return count;
  }
};

class SolutionHamming {
public:
  int hammingWeight(int n) {
    int count = 0;

    while (n != 0) {
      n = (n & (n - 1));
      ++count;
    }

    return count;
  }
};